### PR TITLE
HA groups

### PR DESCRIPTION
**Pull request by gaudenz**
Add functionality to create HA groups. This does not really depend on the Travis CI test fix, but without it the CI does not pass and the tests added in this PR won't even be run.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -409,7 +409,22 @@ pve_cluster_ring0_addr: "{{ ansible_default_ipv4.address }}"
...
 # pve_cluster_bindnet1_addr: "{{ pve_cluster_ring1_addr }}"
+```
 
+You can also configure [HA manager groups][ha-group]:
+```
+pve_cluster_ha_groups: [] # List of HA groups to create in PVE.
+```
+
+This example create a group "lab_node01" for resources assigned to host
+lab-node01:
```
Comment (lae) [inline README.md]: create → creates
host lab-node01 → the lab-node01 host

File: README.md [hunk 2]
```diff
@@ -409,7 +409,22 @@ pve_cluster_ring0_addr: "{{ ansible_default_ipv4.address }}"
...
+```
+pve_cluster_ha_groups:
+  - name: lab_node01
+    comment: "My HA group"
+    nodes: "lab-node01"
```
Comment (lae) [inline README.md]: I think this should probably be another list rather than a string, thoughts?

File: tests/group_vars/all [hunk 3]
```diff
@@ -12,6 +12,10 @@ pve_zfs_zed_email: root@localhost
 pve_ssl_private_key: "{{ lookup('file', ssl_host_key_path) }}"
 pve_ssl_certificate: "{{ lookup('file', ssl_host_cert_path) }}"
 pve_cluster_enabled: yes
+pve_cluster_ha_groups:
+  - name: proxmox_5_01
+    comment: "Resources on proxmox-5-01"
+    nodes: proxmox-5-01
```
Comment (lae) [inline tests/group_vars/all]: and adding to the comment about making this a list - we could add the other 2 proxmox test containers here

Comment (gaudenz) [reply tests/group_vars/all]: OK if we make this a list just to test if having a list with more than one item actually works. I would probably add 2 containers to also test that it does not unconditionally add all of them ;-).

File: README.md (hunk 2 above)
Comment (gaudenz) [reply README.md]: I don't have a strong opinion on this. The way I use HA groups in Proxmox, having a group assigned to multiple nodes does not make much sense. I use them to assign resources to the node where they run under normal conditions (no maintenance, no hardware failure). In the current implementation this is just an opaque string used as an argument when creating the group. This keeps the implementation simple. On the other hand having this as a list would make it more strict. In the end it's probably just a matter of adding a "join".

**Comment by gaudenz** [comment]
Instead of converting this to a module I added a second tasks to update group properties if they don't match. I also changed the `run_once: True` condition to a `when:` condition and added if statements around optional command parameters as requested in the other pull requests.

**Comment by gaudenz** [comment]
@lae Is this now OK or would you like more changes?

**Comment by lae** [comment]
Sorry for putting this off for so long. I was still kind of thinking that the nodes value should a list (as one could make an "HA" inventory group and just pass it) but wanted to check out how it actually worked. I guess since the API returns the nodes as a comma-delimited string the current PR is fine as-is (the following is mainly for my record):

```
vagrant@pve-1:~$ sudo pvesh get /cluster/ha/groups --output=json
[{"comment":"Resources on first cluster host","digest":"d3838b51810de14d10d24475c8fb866172548190","group":"test","nodes":"pve-1","restricted":1,"type":"group"},{"comment":"test2 resources","digest":"fbee3c9903103db28cf7dddbe031e14af67d7cb7","group":"test2","nodes":"pve-1,pve-2","restricted":1,"type":"group"}]
```

**Pull request merged**